src/libraries/Microsoft.Extensions.Http/tests/Microsoft.Extensions.Http.Tests.csproj

```diff
@@ -1,22 +1,35 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>$(DefaultNetCoreTargetFramework);net472</TargetFrameworks>
+    <TargetFrameworks>$(NetCoreAppCurrent);$(NetFrameworkCurrent)</TargetFrameworks>
+    <EnableDefaultItems>true</EnableDefaultItems>
   </PropertyGroup>
 
   <ItemGroup>
-    <Reference Include="Microsoft.Extensions.DependencyInjection.Abstractions" />
-    <Reference Include="Microsoft.Extensions.DependencyInjection" />
-    <Reference Include="Microsoft.Extensions.Http" />
-    <Reference Include="Microsoft.Extensions.Logging.Abstractions" />
-    <Reference Include="Microsoft.Extensions.Logging.Testing" />
-    <Reference Include="Microsoft.Extensions.Logging" />
-    <Reference Include="Microsoft.Extensions.Options" />
-    <Reference Include="Microsoft.Extensions.Primitives" />
+    <Compile Include="$(LibrariesProjectRoot)Microsoft.Extensions.Logging\tests\DI.Common\Common\src\TestSink.cs">
+      <Link>tests\DI.Common\Common\src\TestSink.cs</Link>
+    </Compile>
+    <Compile Include="$(LibrariesProjectRoot)Microsoft.Extensions.Logging\tests\DI.Common\Common\src\WriteContext.cs">
+      <Link>tests\DI.Common\Common\src\WriteContext.cs</Link>
+    </Compile>
+    <Compile Include="$(LibrariesProjectRoot)Microsoft.Extensions.Logging\tests\DI.Common\Common\src\BeginScopeContext.cs">
+      <Link>tests\DI.Common\Common\src\BeginScopeContext.cs</Link>
+    </Compile>
+    <Compile Include="$(LibrariesProjectRoot)Microsoft.Extensions.Logging\tests\DI.Common\Common\src\ITestSink.cs">
+      <Link>tests\DI.Common\Common\src\ITestSink.cs</Link>
+    </Compile>
+    <Compile Include="$(LibrariesProjectRoot)Microsoft.Extensions.Logging\tests\DI.Common\Common\src\TestLogger.cs">
+      <Link>tests\DI.Common\Common\src\TestLogger.cs</Link>
+    </Compile>
+    <Compile Include="$(LibrariesProjectRoot)Microsoft.Extensions.Logging\tests\DI.Common\Common\src\TestLoggerFactory.cs">
+      <Link>tests\DI.Common\Common\src\TestLoggerFactory.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)..\tests\Extensions\SingleThreadedSynchronizationContext.cs" Link="Shared\SingleThreadedSynchronizationContext.cs"/>
   </ItemGroup>
 
   <ItemGroup>
-    <Compile Include="..\..\..\Shared\test\SingleThreadedSynchronizationContext.cs" Link="Shared\SingleThreadedSynchronizationContext.cs"/>
+    <PackageReference Include="Moq" Version="4.10.0" />
```

Comment:
```suggestion
    <PackageReference Include="Moq" Version="$(MoqVersion)" />
```

Reply:
I did this intentionally there is a breaking change with the current Moq version we use. Extensions repo uses 4.10.0

it will cause the two failures below:
```
          Microsoft.Extensions.Http.DefaultHttpClientFactoryTest.Factory_DisposeHandler_DoesNotDisposeInnerHandler [FAIL]
            System.ArgumentException : Method HttpMessageHandler.Dispose is public. Use strong-typed Expect overload instead:
      mock.Setup(x => x.Dispose());

            Stack Trace:
                 at Moq.Protected.ProtectedMock`1.ThrowIfPublicMethod(MethodInfo method, String reflectedTypeName)
                 at Moq.Protected.ProtectedMock`1.Setup(String methodName, Boolean exactParameterMatch, Object[] args)
              C:\CodeHub\runtime\src\libraries\Microsoft.Extensions.Http\tests\DefaultHttpClientFactoryTest.cs(118,0): at Microsoft.Extensions.Http.DefaultHttpClientFactoryTest.<>c.<Factory_DisposeHandler_DoesNotDisposeInnerHandler>b__17_0(HttpMessageHandlerBuilder b)
              C:\CodeHub\runtime\src\libraries\Microsoft.Extensions.Http\src\DefaultHttpClientFactory.cs(195,0): at Microsoft.Extensions.Http.DefaultHttpClientFactory.<>c__DisplayClass17_0.<CreateHandlerEntry>g__Configure|0(HttpMessageHandlerBuilder b)
              C:\CodeHub\runtime\src\libraries\Microsoft.Extensions.Http\src\DefaultHttpClientFactory.cs(177,0): at Microsoft.Extensions.Http.DefaultHttpClientFactory.CreateHandlerEntry(String name)
              C:\CodeHub\runtime\src\libraries\Microsoft.Extensions.Http\src\DefaultHttpClientFactory.cs(107,0): at Microsoft.Extensions.Http.DefaultHttpClientFactory.<>c__DisplayClass14_0.<.ctor>b__1()
              C:\CodeHub\runtime\src\libraries\System.Private.CoreLib\src\System\Lazy.cs(322,0): at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
              C:\CodeHub\runtime\src\libraries\System.Private.CoreLib\src\System\Lazy.cs(346,0): at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
              C:\CodeHub\runtime\src\libraries\System.Private.CoreLib\src\System\Lazy.cs(426,0): at System.Lazy`1.CreateValue()
              C:\CodeHub\runtime\src\libraries\System.Private.CoreLib\src\System\Lazy.cs(507,0): at System.Lazy`1.get_Value()
              C:\CodeHub\runtime\src\libraries\Microsoft.Extensions.Http\src\DefaultHttpClientFactory.cs(144,0): at Microsoft.Extensions.Http.DefaultHttpClientFactory.CreateHandler(String name)
              C:\CodeHub\runtime\src\libraries\Microsoft.Extensions.Http\src\HttpMessageHandlerFactoryExtensions.cs(26,0): at System.Net.Http.HttpMessageHandlerFactoryExtensions.CreateHandler(IHttpMessageHandlerFactory factory)
              C:\CodeHub\runtime\src\libraries\Microsoft.Extensions.Http\tests\DefaultHttpClientFactoryTest.cs(129,0): at Microsoft.Extensions.Http.DefaultHttpClientFactoryTest.Factory_DisposeHandler_DoesNotDisposeInnerHandler()
          Microsoft.Extensions.Http.DefaultHttpClientFactoryTest.Factory_DisposeClient_DoesNotDisposeHandler [FAIL]
            System.ArgumentException : Method HttpMessageHandler.Dispose is public. Use strong-typed Expect overload instead:
      mock.Setup(x => x.Dispose());

            Stack Trace:
                 at Moq.Protected.ProtectedMock`1.ThrowIfPublicMethod(MethodInfo method, String reflectedTypeName)
                 at Moq.Protected.ProtectedMock`1.Setup(String methodName, Boolean exactParameterMatch, Object[] args)
              C:\CodeHub\runtime\src\libraries\Microsoft.Extensions.Http\tests\DefaultHttpClientFactoryTest.cs(93,0): at Microsoft.Extensions.Http.DefaultHttpClientFactoryTest.<>c.<Factory_DisposeClient_DoesNotDisposeHandler>b__16_0(HttpMessageHandlerBuilder b)
              C:\CodeHub\runtime\src\libraries\Microsoft.Extensions.Http\src\DefaultHttpClientFactory.cs(195,0): at Microsoft.Extensions.Http.DefaultHttpClientFactory.<>c__DisplayClass17_0.<CreateHandlerEntry>g__Configure|0(HttpMessageHandlerBuilder b)
              C:\CodeHub\runtime\src\libraries\Microsoft.Extensions.Http\src\DefaultHttpClientFactory.cs(177,0): at Microsoft.Extensions.Http.DefaultHttpClientFactory.CreateHandlerEntry(String name)
              C:\CodeHub\runtime\src\libraries\Microsoft.Extensions.Http\src\DefaultHttpClientFactory.cs(107,0): at Microsoft.Extensions.Http.DefaultHttpClientFactory.<>c__DisplayClass14_0.<.ctor>b__1()
              C:\CodeHub\runtime\src\libraries\System.Private.CoreLib\src\System\Lazy.cs(322,0): at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
              C:\CodeHub\runtime\src\libraries\System.Private.CoreLib\src\System\Lazy.cs(346,0): at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
              C:\CodeHub\runtime\src\libraries\System.Private.CoreLib\src\System\Lazy.cs(426,0): at System.Lazy`1.CreateValue()
              C:\CodeHub\runtime\src\libraries\System.Private.CoreLib\src\System\Lazy.cs(507,0): at System.Lazy`1.get_Value()
              C:\CodeHub\runtime\src\libraries\Microsoft.Extensions.Http\src\DefaultHttpClientFactory.cs(144,0): at Microsoft.Extensions.Http.DefaultHttpClientFactory.CreateHandler(String name)
              C:\CodeHub\runtime\src\libraries\Microsoft.Extensions.Http\src\DefaultHttpClientFactory.cs(125,0): at Microsoft.Extensions.Http.DefaultHttpClientFactory.CreateClient(String name)
              C:\CodeHub\runtime\src\libraries\Microsoft.Extensions.Http\src\HttpClientFactoryExtensions.cs(26,0): at System.Net.Http.HttpClientFactoryExtensions.CreateClient(IHttpClientFactory factory)
              C:\CodeHub\runtime\src\libraries\Microsoft.Extensions.Http\tests\DefaultHttpClientFactoryTest.cs(104,0): at Microsoft.Extensions.Http.DefaultHttpClientFactoryTest.Factory_DisposeClient_DoesNotDisposeHandler()
        Finished:    Microsoft.Extensions.Http.Tests
      === TEST EXECUTION SUMMARY ===
         Microsoft.Extensions.Http.Tests  Total: 70, Errors: 0, Failed: 2, Skipped: 0, Time: 0.922s
```

Reply:
Can you try making the change it suggests or is this a bug in Moq?